lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -355,78 +318,13 @@ public void OnCompleted()
 
         public bool HasChildren()
         {
-            return _children.Count > 0;
-        }
-
-        /// <summary>
-        /// Get a set of nodes containing an incoming message and belonging to candidate set of nodes.
-        /// </summary>
-        /// <param name="nodeSetIdentifier">Candidate set of nodes from which data is to be received</param>
-        /// <returns>A Vector of NodeStruct with incoming data.</returns>
-        private IEnumerable<NodeStruct<T>> GetNodeWithData(IEnumerable<string> nodeSetIdentifier)
-        {
-            CancellationTokenSource timeoutSource = new CancellationTokenSource(_timeout);
-            List<NodeStruct<T>> nodesSubsetWithData = new List<NodeStruct<T>>();
-
-            try
-            {
-                lock (_thisLock)
-                {
-                    foreach (var identifier in nodeSetIdentifier)
-                    {
-                        if (!_idToNodeMap.ContainsKey(identifier))
-                        {
-                            throw new Exception("Trying to get data from the node not present in the node map");
-                        }
-
-                        if (_idToNodeMap[identifier].HasMessage())
-                        {
-                            nodesSubsetWithData.Add(_idToNodeMap[identifier]);
-                        }
-                    }
-
-                    if (nodesSubsetWithData.Count > 0)
-                    {
-                        return nodesSubsetWithData;
-                    }
-
-                    while (_nodesWithData.Count != 0)
```

Comment:
It tires to remove irrelevant nodes such as the parent node from the queue. In the new code, we are not paying attention to those nodes at all.